ikarus/python/dirichletvalues/dirichletvalues.hh

```diff
@@ -20,22 +20,80 @@
 #include <dune/python/pybind11/stl_bind.h>
 
 #include <ikarus/finiteelements/ferequirements.hh>
+
 // PYBIND11_MAKE_OPAQUE(std::vector<bool>);
 namespace Ikarus::Python {
 
+template <class DirichletValues>
+void forwardCorrectFunction(DirichletValues& dirichletValues, const pybind11::function& functor, auto&& cppFunction) {
+  using Basis        = typename DirichletValues::Basis;
+  using Intersection = typename Basis::GridView::Intersection;
+  using BackendType  = typename DirichletValues::BackendType;
+  using MultiIndex   = typename Basis::MultiIndex;
+
+  using LocalViewWrapper = Dune::Python::LocalViewWrapper<Basis>;
+
+  using FixBoundaryDOFsWithGlobalIndexFunction = std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int)>;
+  using FixBoundaryDOFsWithLocalViewFunction =
+      std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&)>;
+  using FixBoundaryDOFsWithIntersectionFunction =
+      std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&, const Intersection&)>;
+
+  // Disambiguate by number of arguments, as casting doesn't properly work with functions
+  pybind11::module inspect_module = pybind11::module::import("inspect");
+  pybind11::object result         = inspect_module.attr("signature")(functor).attr("parameters");
+  size_t numParams                = pybind11::len(result);
+
+  if (numParams == 2) {
+    auto function = functor.template cast<const FixBoundaryDOFsWithGlobalIndexFunction>();
+    auto lambda   = [&](BackendType& vec, const MultiIndex& indexGlobal) {
+      // we explicitly only allow flat indices
+      function(vec.vector(), indexGlobal[0]);
```

Comment:
Ah no this is explicitly simplified here to forward only scalars. For symmetry this should forward the multindex as `indexGlobal` without `[0]`.

Reply:
The returned objects is defined here: [Link](https://gitlab.dune-project.org/staging/dune-functions/-/blob/51661c3085e56f65f491968f7b7a8c7c973a6026/dune/python/functions/globalbasis.hh#L90), therefore the MultiIndex is converted to std::vector<int>.  This should directly work with the bindings.